### PR TITLE
add audit logging when kafka connection fails to initialise

### PR DESCRIPTION
**Pull request by wbittles**
fix for #5028
Signed-off-by: William Bittles <wbittles@uk.ibm.com>

### CODE REVIEW

File: open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnectorAuditCode.java [hunk 1]
```diff
@@ -134,8 +128,13 @@
                     "indicate the cause of this error.  Work to clear the underlying error.  " +
                     "Once fixed, it may be necessary to restart the server to cause a reconnect to Kafka."),
 
-    ;
+    KAFKA_PRODUCER_SEND_ERROR( "OCF-KAFKA-TOPIC-CONNECTOR-0019" ,
+            OMRSAuditLogRecordSeverity.EXCEPTION,
+            " Egeria has encountered an unrecoverable error while attempting to send messages to the following topic {0}",
```
Comment (mandy-chessell) [inline open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnectorAuditCode.java:133]: Specify specific component that is failing - "egeria" is too generic.  My guess is that it should be something like the KafkaOpenMetadataTopicConnector.  Detail the class of exception and any message within it.

File: open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnectorAuditCode.java [hunk 2]
```diff
@@ -134,8 +128,13 @@
...
+            "Ensure that the kakfa service is available",
```
Comment (mandy-chessell) [inline open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnectorAuditCode.java:134]: The system response should describe the state of the server as a result of the failure - is it going to retry this event? shutdown? try to send the next message?

File: open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnectorAuditCode.java [hunk 3]
```diff
@@ -134,8 +128,13 @@
...
+            "Ensure that the kafka service is available"),
```
Comment (mandy-chessell) [inline open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnectorAuditCode.java:135]: For the user action, do they need to do anything to server to restart sends once Kafka is available?

File: open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnectorAuditCode.java [hunk 4]
```diff
@@ -74,19 +74,16 @@
              "The Apache Kafka connector listening on topic {0} received an unexpected exception {1} distributing an event to components within the server.  The event was {2} and the message in the exception was {3}",
              "An incoming event could not be processed by one or more components in the server.",
              "Use the information in the event and the exception message, along with other messages to determine the source of the error."),
-
```
Comment (mandy-chessell) [inline open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnectorAuditCode.java:77]: Why has this whitespace been removed throughout the file?

**Review by mandy-chessell — CHANGES_REQUESTED** [review]

**Comment by wbittles** [comment]
Need to go back to the issue , not sure this fix is correct

**Pull request closed without merging**